File: benchmarks/oopsla25/subjects/291.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f291(const char* s) {
    if (strlen(s) == 0) {
        return;
    }

    if (s[0] == 'a') {
        if (strlen(s) == 1) {
            return;
        } else if (strlen(s) == 2) {
            assert(s[1] == 'b');
        } else {
            exit(1);
        }
    } else {
        assert(s[0] == 'b');
        assert(strlen(s) == 1);
    }
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f291(my_string);
  return 0;
}

File: tests/oopsla25/291.py
#!/usr/bin/env python3

import sys

def f291(s: str):
  if len(s) == 0:
    return
  if s[0] == "a":
    if len(s) == 1:
      return
    elif len(s) == 2:
      assert s[1] == "b"
    else:
      raise Exception
  else:
    assert s[0] == "b"
    assert len(s) == 1

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f291(my_string)
